spec/haproxy/templates/proxies_cidrs.txt_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require 'rspec'
+
+describe 'config/proxies_cidrs.txt' do
+  let(:template) { haproxy_job.template('config/proxies_cidrs.txt') }
+
+  describe 'ha_proxy.proxies_cidrs' do
+    context 'when a space-separated list of cidrs is provided' do
+      it 'has the correct contents' do
+        expect(template.render({
+          'ha_proxy' => {
+            'proxies_cidrs' => '10.0.1.32/27
+                                2600:1f18:4c0b:3207::/64
+          }
+        })).to eq(<<~EXPECTED)
+          # generated from proxies_cidrs.txt.erb
+
+          # BEGIN proxies cidrs
+            10.0.1.32/27
+            2600:1f18:4c0b:3207::/64
+          # END proxies cidrs
+
+        EXPECTED
+      end
+    end
+
+    context 'when a newline-separated, gzipped, base64-encoded list of cidrs is provided' do
+      it 'has the correct contents' do
+        expect(template.render({
+          'ha_proxy' => {
+            'proxies_cidrs' => gzip_and_b64_encode("10.0.1.32/27\n2600:1f18:4c0b:3207::/64")
+          }
+        })).to eq(<<~EXPECTED)
+          # generated from proxies_cidrs.txt.erb
+
+          # BEGIN proxies cidrs
+          10.0.1.32/27
+          2600:1f18:4c0b:3207::/64
```

Comment:
for IPv6 please use IP addresses from the documentation space:
* `2001:db8::/32`
* `3fff::/20`

See [RFC 9637](https://www.rfc-editor.org/rfc/rfc9637.html)

Reply:
thx